### PR TITLE
Initial jobs for sail-operator repo

### PR DESCRIPTION
**Pull request by jwendell**

### CODE REVIEW

File: prow/config/jobs/sail-operator.yaml
Comment (howardjohn) [inline prow/config/jobs/sail-operator.yaml:47]: This gives (very) high level privilege here. This is acceptable, as its a periodic job that runs only on approved+merged code, rather than arbitrary presubmit code. We do this in Istio in many places. However, all Istio code like this is protect by CODEOWNERS where only trusted maintainers have approval access.

So I think this is fine as long as we make sure the repo maintains a similar trust level. In aprticular, do we need the 3 third party Admin tools here https://github.com/istio-ecosystem/sail-operator/settings/access? And will a CODEOWNERS be established? When it is, can we ensure that the owners for the `tools/` is scoped only to Istio maintainers?

File: prow/config/jobs_test.go
Comment (howardjohn) [inline prow/config/jobs_test.go]: case doesn't fallthrough

```suggestion
		case "istio", "istio-ecosystem":
```

File: prow/config.yaml
Comment (howardjohn) [inline prow/config.yaml]: I think this just means "CLA must pass". But we need the CLA to actually run which is owned by LF outside of Istio.

@therealmitchconnors do you know how to get that setup?

Comment (jwendell) [reply prow/config.yaml]: +1. I'll remove it while we get the CI configured at least for the temporary branch.

File: prow/config/jobs_test.go
Comment (jwendell) [reply prow/config/jobs_test.go]: too much C lately :/

**Review by howardjohn — APPROVED** [review]

File: prow/config.yaml
Comment (jwendell) [reply prow/config.yaml]: @therealmitchconnors Do you know how we can get easyCLA installed in istio-ecosystem/sail-operator repo?

**Pull request merged**
